dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/ElasticListener.java

```diff
@@ -228,16 +232,110 @@ private String getNDJSON(List<Entry> publishedEntries) {
+            detachedTool.setNamespace(tool.getNamespace());
+            detachedTool.setRegistry(tool.getRegistry());
+            detachedTool.setPrivateAccess(tool.isPrivateAccess());
+
+            // These are for table
+            detachedTool.setGitUrl(tool.getGitUrl());
+            detachedTool.setName(tool.getName());
+            detachedTool.setToolname(tool.getToolname());
+            detachedEntry = detachedTool;
+        } else if (entry instanceof BioWorkflow) {
+            BioWorkflow bioWorkflow = (BioWorkflow) entry;
+            BioWorkflow detachedBioWorkflow = new BioWorkflow();
+            // These are for facets
+            detachedBioWorkflow.setDescriptorType(bioWorkflow.getDescriptorType());
+            detachedBioWorkflow.setSourceControl(bioWorkflow.getSourceControl());
+            detachedBioWorkflow.setOrganization(bioWorkflow.getOrganization());
+
+            // These are for table
+            detachedBioWorkflow.setWorkflowName(bioWorkflow.getWorkflowName());
+            detachedBioWorkflow.setRepository(bioWorkflow.getRepository());
+            detachedBioWorkflow.setGitUrl(bioWorkflow.getGitUrl());
+            detachedEntry = detachedBioWorkflow;
+        } else {
+            return entry;
+        }
+        detachedEntry.setDescription(entry.getDescription());
+        detachedEntry.setAuthor(entry.getAuthor());
+        detachedEntry.setAliases(entry.getAliases());
+        detachedEntry.setLabels((SortedSet<Label>)entry.getLabels());
+        detachedEntry.setCheckerWorkflow(entry.getCheckerWorkflow());
+        Set<Version> detachedVersions = cloneWorkflowVersion(entry.getWorkflowVersions());
+        detachedEntry.setWorkflowVersions(detachedVersions);
+        entry.getStarredUsers().forEach(user -> detachedEntry.addStarredUser((User)user));
+        String defaultVersion = entry.getDefaultVersion();
+        if (defaultVersion != null) {
+            boolean saneDefaultVersion = detachedVersions.stream().anyMatch(version -> defaultVersion.equals(version.getName()) || defaultVersion.equals(version.getReference()));
+            if (saneDefaultVersion) {
+                // If the tool/workflow has a default version, only keep the default version (and its sourcefile contents and description)
+                Set<Version> newWorkflowVersions = detachedEntry.getWorkflowVersions();
+                newWorkflowVersions.forEach(version -> {
+                    if (!defaultVersion.equals(version.getReference()) && !defaultVersion.equals(version.getName())) {
+                        version.setDescriptionAndDescriptionSource(null, null);
+                        SortedSet<SourceFile> sourceFiles = version.getSourceFiles();
+                        sourceFiles.forEach(sourceFile -> sourceFile.setContent(""));
+                    }
+                });
+            } else {
+                LOGGER.error("Entry has a default version that doesn't exist: " + entry.getEntryPath());
```

Comment:
How many workflows/tools is this an issue for?

Reply:
an issue was made for this already